plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -124,6 +126,11 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle)
             if (summary.getColumnSizes() != null) {
                 Long columnSize = summary.getColumnSizes().get(fieldId);
                 if (columnSize != null) {
+                    if (columnHandle.getBaseType().equals(VarcharType.VARCHAR) || columnHandle.getBaseType().equals(VarbinaryType.VARBINARY)) {
```

Comment:
maybe something like this

``` java
if (columnHandle.getBaseType() instanceof FixedWidthType) {
  // columnSize is the size on disk and Trino column stats is size in memory.
  // The relation between the two is type and data dependent.
  // However, Trino currently does not use data size statistics for fixed-width types
 // (it's not needed for them), so do not report it at all, to avoid reporting some bogus value.
}
else 
```

Reply:
or maybe report columnSize *only* when column type is varchar/varbinary